partcad/src/partcad/project_factory_git.py

```
@@ -86,82 +113,113 @@ def _clone_or_update_repo(self, repo_url, cache_dir=None):
-                            # Need to check for updates
-                            origin.fetch()
-                            repo.git.checkout(self.import_revision, force=True)
-                            origin.pull(force=True, rebase=True)
-                            os.utime(guard_path, (now, now))
+                            # Import a specific revision
+                            if user_config.force_update:
+                                # Ensure "before" doesn't match the desired revision
+                                before = ""
+                            else:
+                                # Read the revision name from the guard file
+                                with open(guard_path, "r") as f:
```

Ruff (0.8.2)

Unnecessary open mode parameters

Remove open mode parameters

(UP015)